### ISSUE TITLE
Cluster: Reload a set of scalers

### ISSUE THREAD

**Issue by ntk148v** [vCloud-DFTBA/faythe#40]
## Expected behavior

When a node joins or leaves, the Autoscaler manager reloads its scalers:
* Force save the current scaler state to Etcd so other nodes can start a scaler correctly.
* Stop a scaler that belongs to another node.
* Kill a scaler.

:crossed_fingers: 

## Actual behavior

When a node join or leaves, Autoscaler manager not only reloads its scalers but also watches Etcd change events. 
* Put a scaler to Etcd -> Put event -> Autoscale manager watches and catches this event, `stopScaler` and `startScaler`.

```golang
// Run starts processing of the autoscale manager
func (m *Manager) Run(ctx context.Context) {
	watch := m.etcdcli.Watch(ctx, model.DefaultScalerPrefix, etcdv3.WithPrefix())
	for {
		select {
		case <-m.stop:
			return
		case watchResp := <-watch:
			if watchResp.Err() != nil {
				level.Error(m.logger).Log("msg", "Error watching cluster state", "err", watchResp.Err())
				break
			}
			for _, event := range watchResp.Events {
				sid := string(event.Kv.Key)
				// Create -> simply create and add it to registry
				if event.IsCreate() {
					m.startScaler(sid, event.Kv.Value)
				}
				// Update -> force recreate scaler
				if event.IsModify() {
					m.stopScaler(sid)
					m.startScaler(sid, event.Kv.Value)
				}
				// Delete -> remove from registry and stop the goroutine
				if event.Type == etcdv3.EventTypeDelete {
					m.stopScaler(sid)
				}
			}
		default:
		}
	}
}
```

* At the same time, a stop scaler request is sent too. Boom, we get panic as consequence :fire: 

```
level=info ts=2019-11-12T08:57:47.227Z caller=cluster.go:67 component=cluster msg="A node is joining to existing cluster..."
level=info ts=2019-11-12T08:57:47.227Z caller=cluster.go:69 component=cluster msg="Use the cluster id to join" id=test2
level=info ts=2019-11-12T08:57:47.234Z caller=main.go:162 msg=Listening address=0.0.0.0:8601
level=info ts=2019-11-12T08:57:54.661Z caller=cluster.go:168 component=cluster msg="A new member is joined" name=VTN-KIENNT65 address=10.61.127.102
level=debug ts=2019-11-12T08:57:54.661Z caller=cluster.go:181 component=cluster msg="The current cluster state" members="map[4f111165872744e8b09329dcabfc51b7:{Name:VTN-KIENNT65 ID:4f111165872744e8b09329dcabfc51b7 Address:10.61.127.102} 5e767ba5764c5ba60e20fab1790ab0ba:{Name:vutuandat-desktop ID:5e767ba5764c5ba60e20fab1790ab0ba Address:10.61.127.51}]"
level=debug ts=2019-11-12T08:57:54.734Z caller=cluster.go:151 component=cluster msg="Renew lease for cluster member" id=7587842299897427734 ttl=14
level=debug ts=2019-11-12T08:58:02.234Z caller=cluster.go:151 component=cluster msg="Renew lease for cluster member" id=7587842299897427734 ttl=14
level=info ts=2019-11-12T08:58:06.166Z caller=manager.go:124 component="autoscale manager" msg="Creating scaler" id=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8
level=debug ts=2019-11-12T08:58:09.734Z caller=cluster.go:151 component=cluster msg="Renew lease for cluster member" id=7587842299897427734 ttl=14
level=error ts=2019-11-12T08:58:16.172Z caller=scaler.go:98 component="autoscale manager" scaler=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8 msg="Executing query failed, skip current interval" query="((node_memory_MemTotal_bytes - (node_memory_MemFree_bytessãasxasxasxob=\"udcntt_cloud_prometheus\"} + node_memory_Cached_bytes{job=\"udcntt_cloud_prometheus\"})) / node_memory_MemTotal_bytes * 100) < 50" err="querying instant: Post http://10.240.201.243:9091/api/v1/query: dial tcp 10.240.201.243:9091: connect: connection refused"
level=debug ts=2019-11-12T08:58:17.234Z caller=cluster.go:151 component=cluster msg="Renew lease for cluster member" id=7587842299897427734 ttl=14
level=debug ts=2019-11-12T08:58:24.734Z caller=cluster.go:151 component=cluster msg="Renew lease for cluster member" id=7587842299897427734 ttl=14
level=error ts=2019-11-12T08:58:26.176Z caller=scaler.go:98 component="autoscale manager" scaler=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8 msg="Executing query failed, skip current interval" query="((node_memory_MemTotal_bytes - (node_memory_MemFree_bytessãasxasxasxob=\"udcntt_cloud_prometheus\"} + node_memory_Cached_bytes{job=\"udcntt_cloud_prometheus\"})) / node_memory_MemTotal_bytes * 100) < 50" err="querying instant: Post http://10.240.201.243:9091/api/v1/query: dial tcp 10.240.201.243:9091: connect: connection refused"
level=debug ts=2019-11-12T08:58:32.234Z caller=cluster.go:151 component=cluster msg="Renew lease for cluster member" id=7587842299897427734 ttl=14
level=info ts=2019-11-12T08:58:33.318Z caller=cluster.go:176 component=cluster msg="A member is left" name=VTN-KIENNT65 address=10.61.127.102
level=debug ts=2019-11-12T08:58:33.318Z caller=cluster.go:181 component=cluster msg="The current cluster state" members="map[5e767ba5764c5ba60e20fab1790ab0ba:{Name:vutuandat-desktop ID:5e767ba5764c5ba60e20fab1790ab0ba Address:10.61.127.51}]"
level=info ts=2019-11-12T08:58:33.319Z caller=manager.go:124 component="autoscale manager" msg="Creating scaler" id=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8
level=info ts=2019-11-12T08:58:33.319Z caller=manager.go:112 component="autoscale manager" msg="Removing scaler" id=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8
level=info ts=2019-11-12T08:58:33.319Z caller=manager.go:112 component="autoscale manager" msg="Removing scaler" id=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8
level=debug ts=2019-11-12T08:58:33.319Z caller=scaler.go:67 component="autoscale manager" scaler=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8 msg="Scaler is stopping" id=b4904bdcd2c9675a8e432898d0311bb8
level=debug ts=2019-11-12T08:58:33.319Z caller=scaler.go:67 component="autoscale manager" scaler=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8 msg="Scaler is stopping" id=b4904bdcd2c9675a8e432898d0311bb8
level=debug ts=2019-11-12T08:58:33.319Z caller=scaler.go:70 component="autoscale manager" scaler=/scalers/eefe6e79585275d3bffd5c7c4ee969d5/b4904bdcd2c9675a8e432898d0311bb8 msg="Scaler is stopped" id=b4904bdcd2c9675a8e432898d0311bb8
panic: close of closed channel

goroutine 30 [running]:
github.com/vCloud-DFTBA/faythe/pkg/autoscaler.(*Scaler).stop(0xc000173560)
	/home/vutuandat/Documents/go/src/github.com/ntk148v/faythe/pkg/autoscaler/scaler.go:68 +0x1b7
github.com/vCloud-DFTBA/faythe/pkg/autoscaler.(*Manager).stopScaler(0xc000336300, 0xc0000ec7d0, 0x4a)
	/home/vutuandat/Documents/go/src/github.com/ntk148v/faythe/pkg/autoscaler/manager.go:113 +0x25b
github.com/vCloud-DFTBA/faythe/pkg/autoscaler.(*Manager).Run(0xc000336300, 0xd90460, 0xc0000c1480)
	/home/vutuandat/Documents/go/src/github.com/ntk148v/faythe/pkg/autoscaler/manager.go:92 +0x354
created by main.main
	/home/vutuandat/Documents/go/src/github.com/ntk148v/faythe/cmd/faythe/main.go:139 +0x153a
exit status 2
```

**Comment by ntk148v** [comment]
@hieulq Here boss.

**Comment by vtdat** [comment]
Fixes in  #34. 

Tested and can close @ntk148v
